app/domain/authentication/authn_k8s/execute_command_in_container.rb

```diff
@@ -11,10 +11,13 @@
 module Authentication
   module AuthnK8s
 
-    KubeExec ||= CommandClass.new(
+    ExecuteCommandInContainer ||= CommandClass.new(
       dependencies: {
-        env:    ENV,
-        logger: Rails.logger
+        env:               ENV,
+        websocket_client:  WebSocket::Client::Simple,
+        message_log_class: MessageLog,
+        websocket_message: WebSocketMessage,
```

Comment:
`WebSocketMessage` shouldn't  be a dependecy, since it's a value object.

Reply:
`MessageLog` is a bit trickier, because I made what I now consider a mistake in its design.  As it is currently, you do need to inject it because it has mutation and raises errors.  However, it should have been designed with an immutable interface (returning copies of itself when altered) and the error raising code should be moved into this class.  

This will make it a simpler value object, and will simplify testing of this class by reducing the number of doubles by one.

I will try to fix this class for you today on a separate branch you can pull in.

Reply:
I have added this fix to my branch.